File: python/src/acedatacloud/resources/kickart.py
"""Kickart e-commerce video resources."""

from __future__ import annotations

from typing import Any, Literal


class Kickart:
    """Synchronous Kickart client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def videos(
        self,
        *,
        duration: Literal[15, 30, 45, 60],
        mode: Literal["fast", "pro"] | None = None,
        type: Literal["intro", "main"] | None = None,
        template_id: str | None = None,
        product_url: str | None = None,
        product_id: str | None = None,
        user_images: list[str] | None = None,
        user_videos: list[str] | None = None,
        aspect_ratio: Literal["9:16", "16:9", "3:4", "4:3", "1:1"] | None = None,
        language: Literal["zh", "en", "en-us", "pt-br", "ja", "es-mx", "id", "ms", "tl"] | None = None,
        purpose: str | None = None,
        prompt: str | None = None,
        nle_subtitle_enabled: bool | None = None,
        use_subtitle_erasure: bool | None = None,
        watermark: bool | None = None,
        callback_url: str | None = None,
        async_: bool | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"duration": duration, **kwargs}
        if mode is not None:
            body["mode"] = mode
        if type is not None:
            body["type"] = type
        if template_id is not None:
            body["template_id"] = template_id
        if product_url is not None:
            body["product_url"] = product_url
        if product_id is not None:
            body["product_id"] = product_id
        if user_images is not None:
            body["user_images"] = user_images
        if user_videos is not None:
            body["user_videos"] = user_videos
        if aspect_ratio is not None:
            body["aspect_ratio"] = aspect_ratio
        if language is not None:
            body["language"] = language
        if purpose is not None:
            body["purpose"] = purpose
        if prompt is not None:
            body["prompt"] = prompt
        if nle_subtitle_enabled is not None:
            body["nle_subtitle_enabled"] = nle_subtitle_enabled
        if use_subtitle_erasure is not None:
            body["use_subtitle_erasure"] = use_subtitle_erasure
        if watermark is not None:
            body["watermark"] = watermark
        if callback_url is not None:
            body["callback_url"] = callback_url
        if async_ is not None:
            body["async"] = async_
        return self._transport.request("POST", "/kickart/videos", json=body)

    def viral_videos(
        self,
        *,
        ref_video: str,
        language: Literal["zh", "en", "en-us", "pt-br", "ja", "es-mx", "id", "ms", "tl"],
        mode: Literal["pro", "advanced"] | None = None,
        template_id: str | None = None,
        product_url: str | None = None,
        product_id: str | None = None,
        product_images: list[str] | None = None,
        model_images: list[str] | None = None,
        ai_product_analysis: bool | None = None,
        similarity: Literal["high", "medium"] | None = None,
        nle_subtitle_enabled: bool | None = None,
        use_subtitle_erasure: bool | None = None,
        prompt: str | None = None,
        location_images: list[str] | None = None,
        watermark: bool | None = None,
        callback_url: str | None = None,
        async_: bool | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"ref_video": ref_video, "language": language, **kwargs}
        if mode is not None:
            body["mode"] = mode
        if template_id is not None:
            body["template_id"] = template_id
        if product_url is not None:
            body["product_url"] = product_url
        if product_id is not None:
            body["product_id"] = product_id
        if product_images is not None:
            body["product_images"] = product_images
        if model_images is not None:
            body["model_images"] = model_images
        if ai_product_analysis is not None:
            body["ai_product_analysis"] = ai_product_analysis
        if similarity is not None:
            body["similarity"] = similarity
        if nle_subtitle_enabled is not None:
            body["nle_subtitle_enabled"] = nle_subtitle_enabled
        if use_subtitle_erasure is not None:
            body["use_subtitle_erasure"] = use_subtitle_erasure
        if prompt is not None:
            body["prompt"] = prompt
        if location_images is not None:
            body["location_images"] = location_images
        if watermark is not None:
            body["watermark"] = watermark
        if callback_url is not None:
            body["callback_url"] = callback_url
        if async_ is not None:
            body["async"] = async_
        return self._transport.request("POST", "/kickart/viral-videos", json=body)

    def template_videos(
        self,
        *,
        template_id: str,
        resource_list: list[dict[str, Any]],
        resolution: str | None = None,
        watermark: bool | None = None,
        callback_url: str | None = None,
        async_: bool | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"template_id": template_id, "resource_list": resource_list, **kwargs}
        if resolution is not None:
            body["resolution"] = resolution
        if watermark is not None:
            body["watermark"] = watermark
        if callback_url is not None:
            body["callback_url"] = callback_url
        if async_ is not None:
            body["async"] = async_
        return self._transport.request("POST", "/kickart/template-videos", json=body)


class AsyncKickart:
    """Async Kickart client."""

    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def videos(
        self,
        *,
        duration: Literal[15, 30, 45, 60],
        mode: Literal["fast", "pro"] | None = None,
        type: Literal["intro", "main"] | None = None,
        template_id: str | None = None,
        product_url: str | None = None,
        product_id: str | None = None,
        user_images: list[str] | None = None,
        user_videos: list[str] | None = None,
        aspect_ratio: Literal["9:16", "16:9", "3:4", "4:3", "1:1"] | None = None,
        language: Literal["zh", "en", "en-us", "pt-br", "ja", "es-mx", "id", "ms", "tl"] | None = None,
        purpose: str | None = None,
        prompt: str | None = None,
        nle_subtitle_enabled: bool | None = None,
        use_subtitle_erasure: bool | None = None,
        watermark: bool | None = None,
        callback_url: str | None = None,
        async_: bool | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"duration": duration, **kwargs}
        if mode is not None:
            body["mode"] = mode
        if type is not None:
            body["type"] = type
        if template_id is not None:
            body["template_id"] = template_id
        if product_url is not None:
            body["product_url"] = product_url
        if product_id is not None:
            body["product_id"] = product_id
        if user_images is not None:
            body["user_images"] = user_images
        if user_videos is not None:
            body["user_videos"] = user_videos
        if aspect_ratio is not None:
            body["aspect_ratio"] = aspect_ratio
        if language is not None:
            body["language"] = language
        if purpose is not None:
            body["purpose"] = purpose
        if prompt is not None:
            body["prompt"] = prompt
        if nle_subtitle_enabled is not None:
            body["nle_subtitle_enabled"] = nle_subtitle_enabled
        if use_subtitle_erasure is not None:
            body["use_subtitle_erasure"] = use_subtitle_erasure
        if watermark is not None:
            body["watermark"] = watermark
        if callback_url is not None:
            body["callback_url"] = callback_url
        if async_ is not None:
            body["async"] = async_
        return await self._transport.request("POST", "/kickart/videos", json=body)

    async def viral_videos(
        self,
        *,
        ref_video: str,
        language: Literal["zh", "en", "en-us", "pt-br", "ja", "es-mx", "id", "ms", "tl"],
        mode: Literal["pro", "advanced"] | None = None,
        template_id: str | None = None,
        product_url: str | None = None,
        product_id: str | None = None,
        product_images: list[str] | None = None,
        model_images: list[str] | None = None,
        ai_product_analysis: bool | None = None,
        similarity: Literal["high", "medium"] | None = None,
        nle_subtitle_enabled: bool | None = None,
        use_subtitle_erasure: bool | None = None,
        prompt: str | None = None,
        location_images: list[str] | None = None,
        watermark: bool | None = None,
        callback_url: str | None = None,
        async_: bool | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"ref_video": ref_video, "language": language, **kwargs}
        if mode is not None:
            body["mode"] = mode
        if template_id is not None:
            body["template_id"] = template_id
        if product_url is not None:
            body["product_url"] = product_url
        if product_id is not None:
            body["product_id"] = product_id
        if product_images is not None:
            body["product_images"] = product_images
        if model_images is not None:
            body["model_images"] = model_images
        if ai_product_analysis is not None:
            body["ai_product_analysis"] = ai_product_analysis
        if similarity is not None:
            body["similarity"] = similarity
        if nle_subtitle_enabled is not None:
            body["nle_subtitle_enabled"] = nle_subtitle_enabled
        if use_subtitle_erasure is not None:
            body["use_subtitle_erasure"] = use_subtitle_erasure
        if prompt is not None:
            body["prompt"] = prompt
        if location_images is not None:
            body["location_images"] = location_images
        if watermark is not None:
            body["watermark"] = watermark
        if callback_url is not None:
            body["callback_url"] = callback_url
        if async_ is not None:
            body["async"] = async_
        return await self._transport.request("POST", "/kickart/viral-videos", json=body)

    async def template_videos(
        self,
        *,
        template_id: str,
        resource_list: list[dict[str, Any]],
        resolution: str | None = None,
        watermark: bool | None = None,
        callback_url: str | None = None,
        async_: bool | None = None,
        **kwargs: Any,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"template_id": template_id, "resource_list": resource_list, **kwargs}
        if resolution is not None:
            body["resolution"] = resolution
        if watermark is not None:
            body["watermark"] = watermark
        if callback_url is not None:
            body["callback_url"] = callback_url
        if async_ is not None:
            body["async"] = async_
        return await self._transport.request("POST", "/kickart/template-videos", json=body)

File: typescript/src/resources/kickart.ts
/** Kickart e-commerce video resources. */

import { Transport } from '../runtime/transport';

export class Kickart {
  constructor(private transport: Transport) {}

  async videos(opts: {
    duration: 15 | 30 | 45 | 60;
    mode?: 'fast' | 'pro';
    type?: 'intro' | 'main';
    templateId?: string;
    productUrl?: string;
    productId?: string;
    userImages?: string[];
    userVideos?: string[];
    aspectRatio?: '9:16' | '16:9' | '3:4' | '4:3' | '1:1';
    language?: 'zh' | 'en' | 'en-us' | 'pt-br' | 'ja' | 'es-mx' | 'id' | 'ms' | 'tl';
    purpose?: string;
    prompt?: string;
    nleSubtitleEnabled?: boolean;
    useSubtitleErasure?: boolean;
    watermark?: boolean;
    callbackUrl?: string;
    async?: boolean;
    [key: string]: unknown;
  }): Promise<Record<string, unknown>> {
    const { duration, mode, type, templateId, productUrl, productId, userImages, userVideos, aspectRatio, language, purpose, prompt, nleSubtitleEnabled, useSubtitleErasure, watermark, callbackUrl, ...rest } = opts;
    const body: Record<string, unknown> = { duration, ...rest };
    if (mode !== undefined) body.mode = mode;
    if (type !== undefined) body.type = type;
    if (templateId !== undefined) body.template_id = templateId;
    if (productUrl !== undefined) body.product_url = productUrl;
    if (productId !== undefined) body.product_id = productId;
    if (userImages !== undefined) body.user_images = userImages;
    if (userVideos !== undefined) body.user_videos = userVideos;
    if (aspectRatio !== undefined) body.aspect_ratio = aspectRatio;
    if (language !== undefined) body.language = language;
    if (purpose !== undefined) body.purpose = purpose;
    if (prompt !== undefined) body.prompt = prompt;
    if (nleSubtitleEnabled !== undefined) body.nle_subtitle_enabled = nleSubtitleEnabled;
    if (useSubtitleErasure !== undefined) body.use_subtitle_erasure = useSubtitleErasure;
    if (watermark !== undefined) body.watermark = watermark;
    if (callbackUrl !== undefined) body.callback_url = callbackUrl;
    return this.transport.request('POST', '/kickart/videos', { json: body });
  }

  async viralVideos(opts: {
    refVideo: string;
    language: 'zh' | 'en' | 'en-us' | 'pt-br' | 'ja' | 'es-mx' | 'id' | 'ms' | 'tl';
    mode?: 'pro' | 'advanced';
    templateId?: string;
    productUrl?: string;
    productId?: string;
    productImages?: string[];
    modelImages?: string[];
    aiProductAnalysis?: boolean;
    similarity?: 'high' | 'medium';
    nleSubtitleEnabled?: boolean;
    useSubtitleErasure?: boolean;
    prompt?: string;
    locationImages?: string[];
    watermark?: boolean;
    callbackUrl?: string;
    async?: boolean;
    [key: string]: unknown;
  }): Promise<Record<string, unknown>> {
    const { refVideo, language, mode, templateId, productUrl, productId, productImages, modelImages, aiProductAnalysis, similarity, nleSubtitleEnabled, useSubtitleErasure, prompt, locationImages, watermark, callbackUrl, ...rest } = opts;
    const body: Record<string, unknown> = { ref_video: refVideo, language, ...rest };
    if (mode !== undefined) body.mode = mode;
    if (templateId !== undefined) body.template_id = templateId;
    if (productUrl !== undefined) body.product_url = productUrl;
    if (productId !== undefined) body.product_id = productId;
    if (productImages !== undefined) body.product_images = productImages;
    if (modelImages !== undefined) body.model_images = modelImages;
    if (aiProductAnalysis !== undefined) body.ai_product_analysis = aiProductAnalysis;
    if (similarity !== undefined) body.similarity = similarity;
    if (nleSubtitleEnabled !== undefined) body.nle_subtitle_enabled = nleSubtitleEnabled;
    if (useSubtitleErasure !== undefined) body.use_subtitle_erasure = useSubtitleErasure;
    if (prompt !== undefined) body.prompt = prompt;
    if (locationImages !== undefined) body.location_images = locationImages;
    if (watermark !== undefined) body.watermark = watermark;
    if (callbackUrl !== undefined) body.callback_url = callbackUrl;
    return this.transport.request('POST', '/kickart/viral-videos', { json: body });
  }

  async templateVideos(opts: {
    templateId: string;
    resourceList: Array<Record<string, unknown>>;
    resolution?: string;
    watermark?: boolean;
    callbackUrl?: string;
    async?: boolean;
    [key: string]: unknown;
  }): Promise<Record<string, unknown>> {
    const { templateId, resourceList, resolution, watermark, callbackUrl, ...rest } = opts;
    const body: Record<string, unknown> = { template_id: templateId, resource_list: resourceList, ...rest };
    if (resolution !== undefined) body.resolution = resolution;
    if (watermark !== undefined) body.watermark = watermark;
    if (callbackUrl !== undefined) body.callback_url = callbackUrl;
    return this.transport.request('POST', '/kickart/template-videos', { json: body });
  }
}
